contributer-jumpstart.md

```diff
@@ -42,6 +42,99 @@ Each Y seconds a cron job (LogTask.java) will reset the data which were used to
-After that, it asks the backend everything it needs to display on the graphs. Then, every Y seconds (see backend section), it asks the backend the newest intervall datas and update its graphs with it.
\ No newline at end of file
+After that, it asks the backend everything it needs to display on the graphs. Then, every Y seconds (see backend section), it asks the backend the newest intervall datas and update its graphs with it.
+
+# Start developing
+## Prior setup
+- First of all, you must have installed JMeter. We will need to have access to its /lib folder, where the jar of the plugins are installed.
+- Install the production plugin by following the instruction at https://www.ubik-ingenierie.com/blog/ubik-load-pack-observability-plugin/ in the Installation section.
+- Once you installed it the normal way, check its name inside the lib folder :
+![pluginJarLocation](screenshot/jmeterJarLocation.png)
+We can see here that the full name of the jar at the time this doc is written is 'ulp-observability-listener-1.0.2.jar'. We will need this name later.
+
+## Front
+```bash
+cd yourPluginPath/ulp-observability-front/src/main/front
+```
+If it is the first time you execute the front :
+```bash
+npm install
+```
+Start the front :
+```bash
+npm start
+```
+At this point you can debug your frontent at localhost:4200.
+You can also see the front at localhost:9090/yourUlpListenerRoute, but you won't have the sources files exposed in the browser, and you would need to re-build the jar everytime as discussed below in the back section.
+
+
+If you need to change the port on which the backend exposes the data, you must change the port in proxy.conf.json.
+
+## Back
+### Make JMeter use your develop plugin
+The backend is a little bit trickier to setup than the frontend. Each time you do a modification to the backend code, you must run the following command to re-build the jar.
+```bash
+mvn clean install
+```
+Once it is done, the jar of the under developpement observability plugin will be located in /back/target :
+![pluginJarLocation](screenshot/jarLocation.png)
+The next step is to replace the jar inside the jmeter libs with this new one. You also need to rename it to match the previous jar name. You can do it either manually or using this command, replacing the args with your own paths and jar names :
+```bash
+mv /home/ubik/Bureau/work/eclipseWorkspace/ulp-observability-plugin/back/target/ulp-observability-listener-1.0.3-SNAPSHOT.jar /home/ubik/Bureau/apache-jmeter-5.5/lib/ulp-observability-listener-1.0.2.jar
+```
+
+After that, your backend change is used by JMeter.
+### Bash script 
+If you are like me, you will be annoyed extremely fast to do everything listed above everytime you change something in the back. We will see here a way to make it faster + how to plug the eclipse debuguer to your JMeter plugin code :
```

Comment:
debugger

Reply:
Replace "If you are like me, you will be annoyed extremely fast to do everything listed above everytime you change something in the back. We will see here "
 by =>  We will see now